File: 241111/스승의 은혜 2/the-grace-form-teacher-2.py
N, B = map(int, input().split())
arr = [int(input()) for _ in range(N)]

T = 0

for i in range(N):
    t, s = 0, 0
    for j in range(N):
        if i == j:
            s += arr[j]/2
        else:
            s += arr[j]
        if s > B:
            break
        t += 1

    T = max(T, t)
print(T)

File: 250122/스승의 은혜 2/the-grace-form-teacher-2.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    // Please write your code here.
    int arr[1000];

    int N, B;
    cin >> N >> B;
    for (int i=0; i<N; i++) 
        cin >> arr[i];

    int max_val = 0;
    for (int i=0; i<N; i++) {

        int cnt = 0;
        int sumCost = 0;
        for (int j=0; j<N; j++) {
            if (i == j) sumCost += arr[j]/2;
            else sumCost += arr[j];

            if (sumCost > B) break;
            cnt++;
        }

        max_val = max(max_val, cnt);
    }
    cout << max_val;

    return 0;
}